apps/gateway/lib/gateway/controllers/auth_controller.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Gateway.Controllers.AuthController do
+  @moduledoc """
+  Controller for users authentication.
+  """
+  use Gateway, :controller
+  alias Gateway.Auth.GoogleTokenManager
+  alias GameBackend.Users
+
+  def validate_token(conn, %{"token_id" => token_id}) do
+    conn = put_resp_content_type(conn, "application/json")
+
+    case GoogleTokenManager.verify_and_validate(token_id) do
+      {:ok, response} ->
+        Users.find_or_create_google_user_by_email(response["email"])
+        send_resp(conn, 200, Jason.encode!(response))
+
+      {:error, error} ->
+        send_resp(conn, 401, Jason.encode!(Map.new(error)))
```

Comment:
I would expect 400 in this case, why 401?

Reply:
It seems more accurate to me since 401 is "unauthorized" but I can change it.

Reply:
But that's the thing, the request is not unauthorized, it happens to be an invalid token (or something). A 401 refers to the authorization to make the request (do we allow this client to call this endpoint/path/resource?) and in this case we allow any client to call us (for now at least)

Based on that a 401 is incorrect (and by extension a 403), from our options remaining it would be either a 400 or a 422 that properly reflect the erro. I think a 400 is good enough because we are likely grouping a bunch of different errors together so it would be generic enough, but if we had details on the specific errors then we can differentiate between a malformed token which would be a 400 and an invalid token which would be a 422

Reply:
Thanks a lot for the explanation, changed!